src/ProjectService.Business/Commands/Project/GetProjectCommand.cs

```diff
@@ -35,49 +37,56 @@ public class GetProjectCommand : IGetProjectCommand
 
         private DepartmentInfo GetDepartment(Guid departmentId, List<string> errors)
         {
-            string errorMessage = $"Can not get department info for DepartmentId '{departmentId}'. Please try again later.";
-
-            DepartmentInfo department = null;
-
             try
             {
-                var departmentResponse = _departmentRequestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>(
-                    IGetDepartmentRequest.CreateObj(null, departmentId)).Result;
-
-                if (departmentResponse.Message.IsSuccess)
+                IOperationResult<IGetDepartmentResponse> departmentResponse = 
+                    _departmentRequestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>
+                    (
+                        IGetDepartmentRequest.CreateObj(null, departmentId)
+                    )
+                    .Result.Message;
+
+                if (departmentResponse.IsSuccess)
                 {
-                    department = _departmentInfoMapper.Map(departmentResponse.Message.Body);
+                    return _departmentInfoMapper.Map(departmentResponse.Body);
                 }
             }
             catch (Exception exc)
             {
-                errors.Add(errorMessage);
-
                 _logger.LogError(exc, "Exception on get department request.");
             }
+            finally
+            {
+                errors.Add($"Can not get department info for DepartmentId '{departmentId}'. Please try again later.");
```

Comment:
Why it is not in "catch" block? In finally we will always get error message, even if there wouldn't any errors, won't we?

Reply:
there is only one case to return success result - departmentResponse.IsSuccess.
other cases are errors

Reply:
you can remove finally here, and add errors after catch. You don't need finally here

Reply:
why? here is 2 places where we can get an errors - it catch and if response is not success. To not duplicate errors.add in 2 places i added finaly.

Reply:
++